### ISSUE TITLE
Use `setup_tb_logging` to log training loss without duplicate computation? 

### ISSUE THREAD

**Issue by isolet** [pytorch/ignite#973]
## ❓ Questions/Help/Support

I'm trying to use `setup_tb_logging` to replace my manual logger setup code. But I can't get `loss` curve in my tensorboard.

The `process_function` of my trainer returns a `loss` with type `torch.tensor`. Looking into `setup_any_logger`, since [no](https://github.com/pytorch/ignite/blob/master/ignite/contrib/engines/common.py#L225) `output_transform` is passed to create `log_handler` so one can [not](https://github.com/pytorch/ignite/blob/master/ignite/contrib/handlers/base_logger.py#L129) get scalar loss values from outputs of `process_function`. And `metric_names="all"` [seems](https://github.com/pytorch/ignite/blob/master/ignite/contrib/handlers/base_logger.py#L116) indicating I should use `Metric`.

In this situation, is attaching `Loss` object to `trainer` the only way to work with `setup_tb_logging`? This might compute loss twice as it is already computed in `process_function`. Thanks!

(As I'm writing this post, I notice maybe I can modify `engine.state.metric` directly after compute loss value in `process_function`. But I'm not sure if I should do that in trainer, I found a `compute_engine` [example](https://pytorch.org/ignite/engine.html#ignite.engine.Engine.add_event_handler) (above this link section)  in the doc though.)

**Comment by vfdev-5** [comment]
@isolet yes, you're right about `setup_tb_logging` behaviour, it logs only metrics and not outputs from `process_function`. 

What can be done :
1) add `OutputHandler` with `output_transform`:
```python
# Attach the logger to the trainer to log training loss at each iteration
tb_logger.attach(trainer,
                 log_handler=OutputHandler(tag="training", output_transform=lambda loss: {'batch loss': loss}),
                 event_name=Events.ITERATION_COMPLETED)
```
2) compute `RunningAverage` metric of batch loss, like it is done with `setup_common_training_handlers`:
https://github.com/pytorch/ignite/blob/68452b987b84116d74896330bd5f75f0aa470bc2/ignite/contrib/engines/common.py#L142

Hope this helps

**Comment by isolet** [comment]
Thanks for the quick reply!

The first way is exactly what I was doing before I found `setup_tb_logging`. I was trying to replace this code by `setup_tb_logging` though.

Thanks for the hint of the your second advice.

Is that safe and recommended to do `engine.state.metrics['loss'] = value` directly before returning in `process_fuction` if I don't want to introduce `Metric` classes to `trainer`?

**Comment by vfdev-5** [comment]
> Is that safe and recommended to do engine.state.metrics['loss'] = value directly before returning in process_fuction if I don't want to introduce Metric classes to trainer?

Yes, it is safe and you can do it. It is almost the same what is done by `RunningAverage` in addition with computing average.

**Comment by isolet** [comment]
Thank you very much! I'll look into it!
